From 28c85250bacd2f256730ba914dbb1375bc0785bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Klitzing?= <aklitzing@gmail.com>
Date: Wed, 29 Jun 2022 09:59:08 +0200
Subject: Provide more details of failed connection

Otherwise the errorString from qtls_openssl is lost that
would help a lot why a tls connection failed.

"Error during SSL handshake: error:1414D17A:SSL routines:tls12_check_peer_sigalg:wrong curve"

Pick-to: 6.4 6.3 6.2
Change-Id: Iab74b73488219686e5926308db02bc4063818c9f
---
 src/network/access/qhttpnetworkconnection.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git x/qtbase/src/network/access/qhttpnetworkconnection.cpp y/qtbase/src/network/access/qhttpnetworkconnection.cpp
index f8ef05a561..1c4e35f741 100644
--- x/qtbase/src/network/access/qhttpnetworkconnection.cpp
+++ y/qtbase/src/network/access/qhttpnetworkconnection.cpp
@@ -925,7 +925,10 @@ QString QHttpNetworkConnectionPrivate::errorDetail(QNetworkReply::NetworkError e
         errorString = QCoreApplication::translate("QHttp", "Unknown protocol specified");
         break;
     case QNetworkReply::SslHandshakeFailedError:
-        errorString = QCoreApplication::translate("QHttp", "SSL handshake failed");
+        if (socket)
+            errorString = QCoreApplication::translate("QHttp", "SSL handshake failed: %1").arg(socket->errorString());
+        else
+            errorString = QCoreApplication::translate("QHttp", "SSL handshake failed");
         break;
     case QNetworkReply::TooManyRedirectsError:
         errorString = QCoreApplication::translate("QHttp", "Too many redirects");
-- 
2.37.0

